backend/app.py

```diff
@@ -216,39 +233,63 @@ def get_cset_members_items(codeset_ids: List[int] = None, con=CON) -> List[Legac
         {'codeset_ids': codeset_ids})
 
 
-def hierarchy(codeset_ids: List[int] = None, selected_concept_ids: List[int] = None) -> Dict:
-    """Get hierarchy of concepts in selected concept sets"""
-    if not selected_concept_ids:
-        selected_concept_ids = get_concept_set_member_ids(codeset_ids, column='concept_id')
-
-    # selected_roots: List[int] = top_level_cids(selected_concept_ids)
+def get_parent_children_map(root_cids: List[int], cids: List[int], con=CON) -> Dict[int, List[int]]:
+    """New hierarchy info from sql"""
+    # int(x): Prevents SQL injection by throwing error if not an integer
```

Comment:
@Sigfried Since I did plain string interpolation on the SQL query, I added this `int(x)` bit that should throw an error and hopefully prevent that.
```python
    # int(x): Prevents SQL injection by throwing error if not an integer
    root_cids: List[int] = [int(x) for x in root_cids]
    cids: List[int] = [int(x) for x in cids]
    root_cids: str = ', '.join([str(x) for x in root_cids])
    cids: str = ', '.join([str(x) for x in cids])
```